routes/dao_coin_exchange_with_fees.go

```diff
@@ -74,23 +72,47 @@ func ValidateTradingFeeMap(feeMap map[string]uint64) error {
 	return nil
 }
 
+func IsExtraDataValueTrueString(extraData map[string]string, key string) bool {
+	newMap := make(map[string][]byte)
+	for k, v := range extraData {
+		newMap[k] = []byte(v)
+	}
+	return IsExtraDataValueTrue(newMap, key)
+}
+
+func IsExtraDataValueTrue(extraData map[string][]byte, key string) bool {
+	if len(extraData) == 0 {
+		return false
+	}
+	val, exists := extraData[key]
+	if exists && len(val) == 1 && val[0] == 1 {
+		return true
+	}
+	return false
+}
+
+func SetExtraDataValueTrue(extraData map[string][]byte, key string) {
+	extraData[key] = []byte{1}
+}
+
+func SetExtraDataValueTrueString(extraData map[string]string, key string) {
+	extraData[key] = string([]byte{1})
+}
+
 func (fes *APIServer) UpdateDaoCoinMarketFees(ww http.ResponseWriter, req *http.Request) {
+	if fes.Config.AmmMetadataPublicKey == "" {
```

Comment:
If there a reason we want this configured at the node level? Could the amm's public key be specified in the request body?